source/shared/cpp/ObjectModel/MarkDownParser.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <iomanip>
+#include <list>
+#include "pch.h"
+#include "BaseCardElement.h"
+#include "Enums.h"
+
+namespace AdaptiveCards
+{
+class MarkDownParser
+{
+public:
+    MarkDownParser (const std::string &txt); 
+
+    enum DelimiterType
+    {
+        Init,
+        Alphanumeric,
+        Puntuation,
+        Escape,
+        WhiteSpace,
+        Underscore = 0x8,
+        Asterisk   = 0x10,
+    };
+
+    enum EmphasisType
+    {
+        EmphasisItalic = 1,
+        EmphasisBold = 2,
+    };
+
+    enum EmphasisState
+    {   
+        // state for the first encounter of emphasis delimiter 
+        EmphasisStart = 0x1,  
+        // state for a emphasis delimiter run 
+        EmphasisRun = 0x2,   
+        InsideEmphasis = EmphasisStart | EmphasisRun,
+        // state for when a emphasis delimiter run is ended 
+        EmphasisEnd = 0x4,       
+        // init state for emphasis detection 
+        EmphasisNone = 0x8,
+        OutsideEmphasis = EmphasisEnd | EmphasisNone,
+        EndOfWords = 0x10,
+        BeginOfWords = 0x20,
+    };
+
+    struct Emphasis
+    {
+        int m_emphCnts;
```

Comment:
>m_emphCnts [](start = 12, length = 10)

Since member visibility defaults to public for structs, struct members usually don't have an m_ prefix. Also, since the struct is called Emphasis, having the emph prefix in this particular member is not necessary. (and let's use the full name instead of cnts)

Reply:
sounds good. thanks

---
In reply to: [154744055](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154744055) [](ancestors = 154744055)